ls1intum/Artemis
# `General`: Search exercises of a course

### Checklist
#### General


#### Server
#### Client


### Motivation and Context



[#5573](https://github.com/ls1intum/Artemis/issues/5573)

### Description


### Steps for Testing

Prerequisites:
- 1 Students
- 1 Course with multiple exercises

1. Log in to Artemis
2. Navigate to Course Overview
3. Open the Course
4. Click the Exercises tab
5. Input the search term
6. Click Enter
7. The exercises should be filtered by the search term

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-exercises.component.ts | 79.14% | 98.86% |

### Screenshots



**Before Search**
<img width="1414" alt="Screen Shot 2022-11-02 at 10 14 53 AM" src="https://user-images.githubusercontent.com/10885503/199472550-bd443850-c793-4319-8e21-c81e4a58c5fd.png">

**After Search**
<img width="1440" alt="Screen Shot 2022-11-02 at 10 15 25 AM" src="https://user-images.githubusercontent.com/10885503/199472589-110605c1-0054-4ce6-92fc-1f593ca2387e.png">

**Mobile**
<img width="282" alt="Screen Shot 2022-11-02 at 10 16 14 AM" src="https://user-images.githubusercontent.com/10885503/199472627-b3f8d024-607f-461d-8eac-d01070890522.png">
